### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by bjankord**
### Summary
Added documentation around terra-icon. 
Moved icon label to display below icon in generated test page.

### CODE REVIEW

File: docs/terra-icon.md [hunk 1]
```diff
@@ -1,9 +1,36 @@
...
 
 ## Getting Started
 
 - Install with [npm](https://www.npmjs.com): `npm install terra-icon`
 - [Download the latest version](https://github.com/cerner/terra-icon/archive/master.zip)
 - Clone the repo: `git clone https://github.com/cerner/terra-icon.git`
+
+## Usage
+
+Terra use a [SVG sprite icon system](https://css-tricks.com/svg-sprites-use-better-icon-fonts/) to display icons.
```
Comment (Matt-Butler) [inline docs/terra-icon.md]: use -> uses

File: docs/terra-icon.md [hunk 2]
```diff
@@ -1,9 +1,36 @@
...
+```
+<svg viewBox="0 0 48 48" class="terra-Icon" aria-hidden="true" role="img">
+  <use xlink:href="#icon-id"></use>
+</svg>
+```
+
+It is recommended to pair icons with visible text which conveys the icon's intended meaning.
+This neatly solves accessibility and i18n issues.
+
+In cases where it is not feasible to display visble text next to icon to convey it's meaning, it is recommended to remove the `aria-hidden="true"` attribute,
```
Comment (Matt-Butler) [inline docs/terra-icon.md]: visble -> visible

**Comment by Matt-Butler** [comment]
+1

File: README.md [hunk 3]
```diff
@@ -5,7 +5,7 @@
 [![Build Status](https://travis-ci.org/cerner/terra-icon.svg?branch=master)](https://travis-ci.org/cerner/terra-icon)
 [![Stories in Ready](https://badge.waffle.io/cerner/terra-ui.com.svg?label=ready&title=Ready)](http://waffle.io/cerner/terra-ui.com)
 
-{insert description}
+The terra-icon component is used to visually represent a literal or symbolic object intended to initiate an action, communicate status, or navigate the workflow.
```
Comment (gneatgeek) [inline README.md]: communicate **a** status

File: docs/terra-icon.md [hunk 4]
```diff
@@ -1,9 +1,77 @@
...
 
 ## Getting Started
 
 - Install with [npm](https://www.npmjs.com): `npm install terra-icon`
 - [Download the latest version](https://github.com/cerner/terra-icon/archive/master.zip)
 - Clone the repo: `git clone https://github.com/cerner/terra-icon.git`
+
+## Usage
+
+Terra uses a [SVG sprite icon system](https://css-tricks.com/svg-sprites-use-better-icon-fonts/) to display icons.
```
Comment (gneatgeek) [inline docs/terra-icon.md]: Terra uses **an** SVG...

File: docs/terra-icon.md [hunk 5]
```diff
@@ -1,9 +1,77 @@
...
+
+Add the [terra-icon SVG sprite](https://github.com/cerner/terra-icon/blob/master/src/sprite.svg?short_path=738f35b) after your project's opening <body> tag:
+This is best done by including this file with your templating engine.
```
Comment (gneatgeek) [inline docs/terra-icon.md]: Whenever possible, include this file with a templating engine.

File: docs/terra-icon.md [hunk 6]
```diff
@@ -1,9 +1,77 @@
...
+
+You can then reference icons in your application with the following markup, where `#icon-id` matches the name of the icon you want to use:
```
Comment (gneatgeek) [inline docs/terra-icon.md]: Use reference icons in the application with the following markup...

File: docs/terra-icon.md [hunk 7]
```diff
@@ -1,9 +1,77 @@
...
+
+
+<svg viewBox="0 0 48 48" class="terra-Icon u-text-danger" aria-hidden="true" role="img">
+  <use xlink:href="#cancel"></use>
+</svg>
+```
+
+### Sizing
+
+By default, all terra icons are styled with a `width` and `height` of `1em`. For cross-browsers sizing, both a width and height are needed.
```
Comment (gneatgeek) [inline docs/terra-icon.md]: Capitalize Terra

File: docs/terra-icon.md [hunk 8]
```diff
@@ -1,9 +1,77 @@
...
+### Color
+
+By default, all icons have a [fill]((https://css-tricks.com/almanac/properties/f/fill/)) styled to [currentColor](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#currentColor_keyword).
+This allows the color of icons to be set by [color property](https://developer.mozilla.org/en-US/docs/Web/CSS/color), similar to how you would color font-icons.
+Specific icons come with predefined colors which should not be changed.
+
+The following code would create an icon colored `#C00`.
+
+```
+<style>
```
Comment (gneatgeek) [inline docs/terra-icon.md]: The following code is an example of how to add color (`#C00`) to an icon using a custom class:

```
.u-text-danger {
  color: #C00;
}
```

**Comment by jpruetting** [comment]
+1

**Pull request merged**